qasp/agile_bpa_qasp.md

```diff
@@ -0,0 +1,115 @@
+towards the final delivered software code. Note that the performance
+requirements listed below are required for the final deliverable.
+However, the sprints and incremental delivery of code will be assessed
+by the Government to ensure that the contractor is on a path to
+successful final delivery.
+
+Deliverable or Required Services Performance Standard(s):
+
+  **Deliverable or Required Services** | **Performance Standard(s)** | **Acceptable Quality Level (AQL)** | **Method of Surveillance**
+   ----------- | ---------------- | -------------- | ---------------
+   Tested Code | Code delivered under the order must have substantial test code coverage and a clean code base. | Minimum of 90% test coverage of all relevant code | Combination of manual review and automated testing, using agreed-upon publicly-available SaaS products
+   Accessible | Client-side rendering must conform with section 508 standards. | 0 errors reported for 508 Standards using an automated scanner and 0 errors reported in manual testing |  http://squizlabs.github.io/HTML_CodeSniffer/ or https://github.com/pa11y/pa11y
+   Deployed | Code must successfully build and deploy into staging environment. | Successful build with a single command | Combination of manual review and automatic testing
+   Documented | All dependencies (and licenses for dependencies) are listed and all major functions are documented. | All dependencies are listed and the licenses are documented. Software/source code is documented. System diagram is provided. | Combination of manual review and automatic testing
+   Available | Code must be stored in a version-controlled open-source repository. | All of the code needed to run the front end of the prototype must be available. | 18F will assess code availability.
+   User research | Usability testing and other user research methods must be conducted at regular intervals throughout the development process (not just at the beginning or end). |  Artifacts from usability testing and/or other research methods with end-users are available at the end of every applicable sprint, in accordance with the vendor’s research plan. | 18F will evaluate the artifacts based on a research plan provided by the vendor at the end of the second sprint.
+   Secure | Code must be free of medium- and high-level static and dynamic security vulnerabilities | Clean tests from a static testing SaaS, such as Gemnasium, and from OWASP ZAP, and/or documentation explaining any false positives. |  https://pages.18f.gov/before-you-ship/
+
+PROCEDURES
+----------
+
+The COR/ACOR, along with the 18F Product Lead and the product owner,
+will inspect all tasks required by the call order to ensure contractor
+compliance with the call order requirements at the conclusion of each
+sprint, which shall have a length of two weeks or less. Delivery will
+occur by pull request from the contractor’s repository to the 18F
+repository. If inspection results are satisfactory, the pull request
```

Comment:
That's the path we took with 14c.  I think it's good to have that flexibility, but we have to be careful about accepting code that isn't conforming.  I accepted code with low code coverage and that ended up being a major sticking point later.  (The vendor did a good job with it, though, it was just unpleasant how it turned into a whole big thing.)

I also accepted a hard-coded configuration _after_ I had asked the vendor to make it an external configuration (they switched it from an unacceptable value to a more acceptable one).  Then a few weeks later we find out that the partner can't run it with the "more acceptable" configuration and need the vendor to change it.  I should have stuck with the 12-factor rules, but I felt like it was fairly minor in the grand scheme.

All said, I wish I had been more strict from the first sprint.  I also wish I had spent more time talking to the vendor early on to get on the same page about expectations.